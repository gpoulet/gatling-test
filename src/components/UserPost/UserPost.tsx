import React, {FC} from "react";
import {Post} from "../../models";
import Card from "react-bootstrap/Card";

export type UserPostProps = {
    post: Post
}

const UserPost: FC<UserPostProps> = ({post}) => {
    const {id, title, body} = post;
    return (
        <Card className="userpost-card">
            <Card.Header>Post #{id}</Card.Header>
            <Card.Body>
                <Card.Title>{title}</Card.Title>
                <Card.Text>
                    {body}
                </Card.Text>
            </Card.Body>
        </Card>
    )
}

export default UserPost