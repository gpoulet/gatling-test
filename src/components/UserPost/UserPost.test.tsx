import React from "react";
import renderer from "react-test-renderer";
import UserPost from "./UserPost";

const post1 = {
  userId: 1,
  id: 1,
  title: "titre 1",
  body: "lorem",
};

it("UserPost renders correctly", () => {
  const tree = renderer.create(<UserPost post={post1} />).toJSON();
  expect(tree).toMatchSnapshot();
});
