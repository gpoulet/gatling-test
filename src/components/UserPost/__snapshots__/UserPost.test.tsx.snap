// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPost renders correctly 1`] = `
<div
  className="userpost-card card"
>
  <div
    className="card-header"
  >
    Post #
    1
  </div>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      titre 1
    </div>
    <p
      className="card-text"
    >
      lorem
    </p>
  </div>
</div>
`;
